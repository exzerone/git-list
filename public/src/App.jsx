import React, { useState, useEffect } from 'react';
import Axios from 'axios';
import CardList from './components/CardList.jsx'


function App() {
	let [users, setUsers] = useState([]);

	useEffect(()=> {
		Axios.get('https://api.github.com/users', {
			params: {
				since: 0
			}
		})
		.then(data => {
			if (data) {
				setUsers(data.data);
			}
		})
		.catch(error => console.log(error))
	}, [])

	return (
		<div>
			<h1 className="title">
					Github Users
			</h1>
			<CardList list={users}/>
		</div>
	)
}

export default App;
