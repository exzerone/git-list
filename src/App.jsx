import React, { useState, useEffect } from 'react';
import CardList from './components/CardList.jsx';
import Pagination from './components/Pagination.jsx';
import { getUsers, getUserDetail } from './github/index';
import UserDetail from './components/UserDetail.jsx';
import styled from 'styled-components';

const Title = styled.h1`
	text-align: center;
	background-color: #b6ffdf;
	color: black;
	padding: 25px;
	position: relative;
	top: 1px;
	margin-top: 0px;
	border-radius: 25px;
`;

const ContentContainer = styled.div`
	padding: 15px;
`;

const Loading = styled.div`
	font-size: 16px;
	margin: 30px 0;
`;

 function App() {
	const [users, setUsers] = useState([]);
	const [page, setPage] = useState(1);
	const [isLoading, setIsLoading] = useState(false);
	const [showModal, setShowModal] = useState(false);
	const [userDetail, setUserDetail] = useState({});

	useEffect(() => {
		getUsers(30 * page - 30, 30, setIsLoading, setUsers);
	}, [page]);

	const handlePageClick = (page) => {
		setPage(page);
	};

	const handleUserClick = (username) => {
		getUserDetail({ username, setShowModal, setUserDetail });
	};

	return (
		<div id="App">
			<Title className="title">Github Users List</Title>
			<ContentContainer className="content_container">
				{isLoading ? (
					<Loading className="loading">Loading...</Loading>
				) : (
					<Pagination
						totalPages={20}
						handleClick={handlePageClick}
						page={page}
					/>
				)}
				<CardList list={users} handleUserClick={handleUserClick} />
				{showModal ? (
					<UserDetail
						handleClose={() => {
							setShowModal(false);
						}}
						{...userDetail}
					/>
				) : null}
			</ContentContainer>
		</div>
	);
}

export default App;
