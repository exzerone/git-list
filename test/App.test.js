import React from 'react';
import { shallow } from 'enzyme';
import App from '../src/App.jsx';

let wrapper = shallow(<App/>);

describe('App', ()=>{
	it("should render without crashing", ()=>{
		expect(wrapper).toMatchSnapshot();
	});

	it('should render title of the site correctly', ()=>{
		expect(wrapper.find('.title').text()).toEqual('Github Users List')
	});
});