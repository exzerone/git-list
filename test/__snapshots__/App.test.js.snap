// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render without crashing 1`] = `
<div
  id="App"
>
  <styled.h1
    className="title"
  >
    Github Users List
  </styled.h1>
  <styled.div
    className="content_container"
  >
    <h2>
      Number Of Users Per Page: 
      0
    </h2>
    <Pagination
      handleClick={[Function]}
      page={1}
      totalPages={20}
    />
    <CardList
      list={Array []}
    />
  </styled.div>
</div>
`;
